core/templates/dev/head/domain/question/QuestionPlayerBackendAPIService.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Service to receive questions for practice given a set of
+ * skill_ids.
+ */
+oppia.constant(
+  'QUESTION_PLAYER_URL_TEMPLATE',
+  '/question_player_handler?skill_ids=<skill_ids>&question_count' +
+  '=<question_count>&start_cursor=<start_cursor>');
+
+oppia.factory('QuestionPlayerBackendApiService', [
+  '$http', '$q', 'UrlInterpolationService', 'QUESTION_PLAYER_URL_TEMPLATE',
+  function($http, $q, UrlInterpolationService, QUESTION_PLAYER_URL_TEMPLATE) {
+    var _startCursor = '';
+    var _fetchQuestions = function(
+        skillIds, questionCount, successCallback, errorCallback) {
+      if (!_inputValidation(skillIds, questionCount, errorCallback)) {
+        return;
+      }
+      var questionDataUrl = UrlInterpolationService.interpolateUrl(
+        QUESTION_PLAYER_URL_TEMPLATE, {
+          skill_ids: skillIds,
+          question_count: questionCount,
+          start_cursor: _startCursor
+        });
+
+      $http.get(questionDataUrl).then(function(response) {
+        var questionDicts = angular.copy(response.data.question_dicts);
+        _startCursor = response.data.next_start_cursor;
+        if (successCallback) {
+          successCallback(questionDicts);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    /**
+     * Does basic validation on input
+     */
+    var _inputValidation = function(skillIds, questionCount, errorCallback) {
+      if (!skillIds) {
+        errorCallback('Skill ids value is expected but is missing');
+        return false;
+      }
+
+      if (isNaN(questionCount)) {
+        errorCallback('Question count has to be a number');
+        return false;
+      }
+
+      if (parseInt(questionCount) < 0) {
```

Comment:
Also, shouldn't we just reject if questionCount isn't an int? We should not need to do parseInt() conversion here.

Reply:
yes, that is why there is a check earlier to check if its a number or not. If its not a number, a false value is returned

Reply:
ah I see what you mean. The reason I have done that is if the frontend passes a string for count, then this would take care of that. 
@vinitamurthi will we pass it as an int from frontend or will it be passed as a string?

Reply:
At this point it should still be an int. Only do string conversion at the boundary of the application (i.e. just before the request is made).

Reply:
done

Reply:
Yes as sean mentioned, I think its good to pass an int itself to the API service, we can convert to string right before sending out the request